roles/nf-core/tasks/main.yml

```diff
@@ -0,0 +1,72 @@
+---
+
+- name: install nf-core
+  local_action:
+    module: pip
+    virtualenv_command: /lupus/ngi/irma3/virtualenv-15.0.0/virtualenv.py
+    virtualenv: "{{ nf_core_env }}"
+    name: nf-core
+    state: present
+
+- name: Download NextFlow
+  local_action:
+    get_url url="{{ nextflow_download_url }}" dest="{{ nf_core_env }}/bin" mode="u+rwx,g=rwx"
+
+- name: Install NextFlow
+  local_action:
+    module: command
+    _raw_params: "{{ nf_core_env }}/bin/nextflow"
+  environment:
+    NXF_LAUNCHER: "/scratch"
+    NXF_HOME: "{{ nf_core_env }}/nextflow"
+    NXF_WORK: "/scratch"
+
+- name: nf-core
+  command: "{{ nf_core_env }}/bin/nf-core -v download {{ item.name }} --singularity --compress none --outdir {{ sw_path }}{{ item.name }} --release {{ item.release }}"
+  environment:
+    PATH: "{{ ansible_env.PATH }}:{{ nf_core_env }}/bin"
+  args:
+    chdir: "{{ sw_path }}"
+  with_items: "{{ pipelines }}"
+
+- name: Create directories for singularity images
+  file:
+    path: "{{ ngi_containers }}/{{ item.name }}"
+    state: directory
+    mode: g+s
+  with_items: "{{ pipelines }}"
+
+- name: Move singularity images
+  shell: "mv {{ sw_path }}{{ item.0.name }}/singularity-images/{{ item.1 }} {{ ngi_containers }}/{{ item.0.name }}/"
```

Comment:
Why do we need to move the singularity images?

Reply:
All the singularity images were put into a common directory in the original flow, so we just mimicked it here

Reply:
`nf-core download` pulls the singularity images and organises the files into a directory structure along with a config so that if you run the workflow from the pipeline it knows where to find the singularity image without needing a `-with-singularity <path>` command line argument.

Unless we have a specific reason for moving the singularity images, it seems that it would be better to leave them where they are..?

Reply:
Apologies, the config to no longer need `-with-singularity <path>` is not yet implemented. See https://github.com/nf-core/tools/issues/464